src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs

```diff
@@ -132,25 +131,59 @@ void TrySetContextualProperties(object? instance, XamlObjectDefinition control)
 				}
 			}
 
-			if (createInstanceFromXClass && TypeResolver.FindType(classMember?.Value?.ToString()) is { } classType)
+			if (createInstanceFromXClass && TypeResolver.FindTypeByXClass(control) is { } classType)
 			{
 				var created = Activator.CreateInstance(classType);
 				TrySetContextualProperties(created, control);
 
 				return created;
 			}
 
+			var type = TypeResolver.FindType(control.Type);
+#if false
```

Comment:
Why is this disabled? Is it only for the comments?

Reply:
It was an attempt to impl the rule. But it doesnt seem to match the observation, hence why I disabled it.